server/channels/api4/file.go

```diff
@@ -490,14 +490,14 @@ func getFile(c *Context, w http.ResponseWriter, r *http.Request) {
 		return
 	}
 	perm := c.App.SessionHasPermissionToReadChannel(c.AppContext, *c.AppContext.Session(), channel)
-	if info.CreatorId == model.BookmarkFileOwner {
-		if !perm {
+	if !c.IsSystemAdmin() && !perm {
+		if info.CreatorId == model.BookmarkFileOwner {
+			c.SetPermissionError(model.PermissionReadChannelContent)
+			return
+		} else if info.CreatorId != c.AppContext.Session().UserId {
 			c.SetPermissionError(model.PermissionReadChannelContent)
 			return
```

Comment:
This feels overly complex, and I'm not clear on what if `info.CreatorId == model.BookmarkFileOwner` means in this context -- why should it matter to the current user's permissions if the creator is the owner?

If the rule is that sysadmins should always have permissions, let's roll that into the perm value. 
We have these conditions, (correct me if I'm wrong):
- is sysAdmin: always has permissions
- is creator of the bookmark: has permissions even if they don't have channel permissions -- effectively always has permissions, right?
- has channel read permissions: has permissions

so, `perm = isSysadmin || isCreator || hasChannelPermissions`
right?

Reply:
I am not familiar myself with bookmarks myself, so I cannot comment on that one (it also feels obscure to me).

I just tried to factor in the "IsSystemAdmin" with just a bit of refactoring.

But I like your idea, let me take a stab at it.